apps/api/src/domain/models/digital-text/commands/add-audio-for-digital-text-page/add-audio-for-digital-text-page.command-handler.ts

```diff
@@ -0,0 +1,75 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { CommandHandler } from '@coscrad/commands';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot, ResourceType } from '../../../../../domain/types/ResourceType';
+import { InternalError, isInternalError } from '../../../../../lib/errors/InternalError';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { AudioItem } from '../../../audio-item/entities/audio-item.entity';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent, IEventPayload } from '../../../shared/events/base-event.entity';
+import { EventRecordMetadata } from '../../../shared/events/types/EventRecordMetadata';
+import { DigitalText } from '../../entities';
+import { AddAudioForDigitalTextPage } from './add-audio-for-digital-text-page.command';
+import { AudioAddedForDigitalTextPage } from './audio-added-for-digital-text-page.event';
+
+@CommandHandler(AddAudioForDigitalTextPage)
+export class AddAudioForDigitalTextPageCommandHandler extends BaseUpdateCommandHandler<DigitalText> {
+    protected actOnInstance(
+        digitalText: DigitalText,
+        { audioItemId, pageIdentifier, languageCode }: AddAudioForDigitalTextPage
+    ): ResultOrError<DigitalText> {
+        /**
+         * It's easy to mix up the audio item ID and the page identifier in this
```

Comment:
Realistically, we just need to rely on the test and use the inline docs. But Let's put a little thought into it. If we think we will add more parameters later, it might be a good call to use a single object.

Reply:
This API is not going to change. Let's just be sure to think carefully about the order of params.